packages/contracts/contracts/registryManager/RegistryManager.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { ICommon } from "../interfaces/ICommon.sol";
+import { IRecipientRegistry } from "../interfaces/IRecipientRegistry.sol";
+import { IRegistryManager } from "../interfaces/IRegistryManager.sol";
+
+/// @title RegistryManager
+/// @notice Contract that allows to use send, approve, reject requests to RecipientRegistry.
+contract RegistryManager is Ownable, IRegistryManager, ICommon {
+  /// @notice requests
+  mapping(uint256 => Request) internal requests;
+
+  /// @inheritdoc IRegistryManager
+  uint256 public requestCount;
+
+  /// @notice Initialize registry manager
+  constructor() payable Ownable(msg.sender) {}
+
+  /// @notice Check if request is valid
+  modifier isValidRequest(Request calldata request) {
+    // TODO: check registry owner as well to make sure it's a part of maci
+    if (request.registry == address(0)) {
+      revert ValidationError();
+    }
+
+    if (request.recipient.recipient == address(0)) {
+      revert ValidationError();
+    }
+
+    uint256 count = IRecipientRegistry(request.registry).recipientCount();
+    bool withIndex = request.requestType == RequestType.Change || request.requestType == RequestType.Remove;
+
+    if (request.index >= count && withIndex) {
+      revert ValidationError();
+    }
+
+    _;
+  }
+
+  /// @notice Check if request is pending and exists
+  /// @param index Request index
+  modifier isPending(uint256 index) {
+    if (index >= requestCount || requests[index].status != Status.Pending) {
+      revert OperationError();
+    }
+
+    _;
+  }
+
+  /// @inheritdoc IRegistryManager
+  function process(Request calldata request) public virtual override isValidRequest(request) {
+    requests[requestCount] = request;
+    requests[requestCount].status = Status.Pending;
+
+    unchecked {
+      requestCount++;
+    }
+
+    emit RequestSent(
+      request.registry,
+      request.requestType,
+      request.recipient.recipient,
+      request.index,
+      request.recipient.id,
+      request.recipient.metadataUrl
+    );
+  }
+
+  /// @inheritdoc IRegistryManager
+  function approve(uint256 index) public virtual override onlyOwner isPending(index) {
+    Request memory request = requests[index];
+    IRecipientRegistry registry = IRecipientRegistry(request.registry);
+
+    if (request.requestType == RequestType.Change) {
+      approveRequest(index, request);
```

Comment:
should we remove the internal function and move over here as it's not used elsewhere (or maybe im not seeing it)? we could remove the 3 calls to it and put the logic after the if/else block

Reply:
also im not sure it's actually being updated as request is memory not storage